simulators/ethereum/rpc/ethclient.go

```diff
@@ -206,6 +206,8 @@ func balanceAndNonceAtTest(t *TestEnv) {
 	exp := new(big.Int).Set(sourceAddressBalanceBefore)
 	exp.Sub(exp, amount)
 	exp.Sub(exp, new(big.Int).Mul(big.NewInt(int64(receipt.GasUsed)), valueTx.GasPrice()))
+	// subtract citrea diff size * l1 fee rate
+	exp.Sub(exp, new(big.Int).Mul(big.NewInt(104), big.NewInt(10)))
```

Comment:
can you get this from transaction receipt (diffSize) and eth_getBlockByNumber (l1FeeRate)?